src/src/PluginStructs/P039_data_struct.h

```diff
@@ -0,0 +1,38 @@
+#ifndef PLUGINSTRUCTS_P039_DATA_STRUCT_H
+#define PLUGINSTRUCTS_P039_DATA_STRUCT_H
+
+#include "../../_Plugin_Helper.h"
+
+#ifdef USES_P039
+
+
+struct P039_data_struct : public PluginTaskData_base {
+public:
+
+  P039_data_struct(uint8_t                mainState,
+                   uint8_t                command,
+                   uint16_t               conversionResult,
+                   uint8_t                deviceFaults,
+                   unsigned long          timer,
+                   bool                   convReady);
+
```

Comment:
Please also add a default constructor (this line is only needed in the .h file), since you already initialize the members to a proper default.

```c++
P039_data_struct() = default;
```

Reply:
I am a "plain old C" guy and not so firm with C++ - thank you for the remark and hint, helps a lot ! Solved using your proposal within the next push.

Reply:
Well be prepared for a really quick crash course C++ then :)
A lot of the newer code in ESPEasy is really object-oriented, so you may have to get familiar with OO to grasp it.

The analogy that made me finally understand OO was quite simple.

Traditional programming is like baking a cake following a receipe.
- Add ingredients
- Mix it all
- pour in a tin
- put in the oven.

Object oriented programming:
- Construct an object: `Cake cake;
- Call `cake.bake_yourself(receipe);`
- `exit 0;`

Reply:
@TD-er : preparing a push to solve your findings - most things done, but i stick to my personal vow -> do not commit at the end of a day, unless you want to redo everything the next morning...

Reply:
Solved in latest push !